template.php

```diff
@@ -150,6 +150,17 @@ function campaignion_foundation_preprocess_block(&$vars) {
   }
 }
 
+/**
+ * Prepares variables for field templates.
+ */
+function campaignion_foundation_preprocess_field(&$vars) {
+  $element = &$vars['element'];
+  // Additional theme hook for pgbars displayed on teasers.
+  if ($element['#field_type'] == 'pgbar' && $element['#view_mode'] == 'teaser') {
+    array_unshift($vars['items'][0]['#theme'], 'pgbar__teaser');
```

Comment:
This will probably break the [template based theme hooks](https://github.com/moreonion/pgbar/blob/821bae0e58c42538862474b96fcc562d584b0511/fields.inc.php#L270) (eg. thermometer style pgbars): If there is a `pgbar--teaser.tpl.php` this will override whatever comes after it in the array.

Probably something like …

```php
$additional = [];
foreach ($vars['items'][0]['#theme'] as $hook) {
  $additional[] = $hook . '__teaser';
}
$vars['items'][0]['#theme'] = array_merge($additional, $vars['items'][0]['#theme');
```
… would be better. I’m not sure whether `hook_field_attach_view_alter()` wouldn’t be more appropriate to handle this. But if in doubt leave it here.

Reply:
That's on purpose, the teaser template should win over thermometer ;)

I've copied the hook from [Simplicity](https://github.com/moreonion/simplicity/blob/7.x-1.x/template.php#L9), can give `hook_field_attach_view_alter()` a try too.

Reply:
Ok, then never mind. I was assuming it shouldn’t override the other styles.

Reply:
This seems to work as well:

```php
/**
 * Implements hook_field_attach_view_alter().
 *
 * Add an additional theme hook for pgbars displayed on teasers.
 */
function campaignion_foundation_field_attach_view_alter(&$output, $context) {
  if (!empty($output['pgbar_default']) && $context['view_mode'] == 'teaser') {
    array_unshift($output['pgbar_default'][0]['#theme'], 'pgbar__teaser');
  }
}
```

Which function should we use?

Reply:
I think I prefer the `hook_attach_view_alter()` … and could you add a comment that says that overriding the pgbar__$template is intentional? Then it’s perfect.

Thanks for trying out the other hook!